src/search/merge_and_shrink/transition_system.h

```diff
@@ -94,51 +93,50 @@ class TransitionSystem {
     const int num_variables;
     std::vector<int> incorporated_variables;
 
+    const Labels &labels;
     /*
       All locally equivalent labels are grouped together, and their
-      transitions are only stored once for every such group, see below.
-
-      LabelEquivalenceRelation stores the equivalence relation over all
-      labels of the underlying factored transition system.
-    */
-    std::unique_ptr<LabelEquivalenceRelation> label_equivalence_relation;
-
-    /*
-      The transitions of a label group are indexed via its ID. The ID of a
-      group does not change, and hence its transitions are never moved.
-
-      We tested different alternatives to store the transitions, but they all
-      performed worse: storing a vector transitions in the label group increases
-      memory usage and runtime; storing the transitions more compactly and
-      incrementally increasing the size of transitions_of_groups whenever a
-      new label group is added also increases runtime. See also issue492 and
-      issue521.
+      transitions are only stored once for every such group. Each such group
+      is represented by a local label (LocalLabelInfo). Local labels can be
+      mapped back to the set of labels they represent. Their cost is
+      the minimum cost of all represented labels.
     */
-    std::vector<std::vector<Transition>> transitions_by_group_id;
+    std::vector<int> label_to_local_label;
+    std::vector<LocalLabelInfo> local_label_infos;
 
     int num_states;
     std::vector<bool> goal_states;
     int init_state;
 
     /*
-      Check if two or more labels are locally equivalent to each other, and
-      if so, update the label equivalence relation.
+      Check if two or more local labels are locally equivalent to each other,
+      and if so, merge them and store their transitions only once.
```

Comment:
Is "locally equivalent" necessary, or would "equivalent" fit better? Is there a meaningful other (non-local) way in which "local labels" can be equivalent?

Reply:
(If not, perhaps the method name might then also change to `compute_equivalent_local_labels` or some such.)

If you prefer the comment in the current form, we can leave it as is.